packages/@lwc/engine-core/src/framework/decorators/api.ts

```diff
@@ -151,22 +151,26 @@ export function createPublicAccessorDescriptor(
                 );
             }
             if (set) {
-                if (features.ENABLE_REACTIVE_SETTER) {
-                    let ro = vm.oar[key as any];
-                    if (isUndefined(ro)) {
-                        ro = vm.oar[key as any] = new AccessorReactiveObserver(vm, set);
+                // On the server side, we don't need mutation tracking. Skipping it improves performance.
+                // Note the `if` cannot be simplified because `features.ENABLE_REACTIVE_SETTER` must be transformed
+                if (process.env.IS_BROWSER) {
```

Comment:
While going through this PR, I thought why does the `if (process.env.IS_BROWSER)` branching have to be in the utils(like mutation-tracker.ts) instead of letting the caller deciding if something needs to be reactive or not. But checking for references of the function shows there are many places that these functions are used and it can get complex if we start adding branching logic there. So for readability, it is best we branch it in the util.

Now for consistency, `AccessorReactiveObserver` should follow the same pattern as `createReactiveObserver` and let the dummy be stubbed out based on `process.env.IS_BROWSER`.

Reply:
That sounds reasonable. Although I wonder if we could remove this code entirely instead? Do we ever plan to enable `ENABLE_REACTIVE_SETTER`?

Reply:
OK, added `createAccessorReactiveObserver`.